### PR TITLE
Added script builder and adapter for new storage provider (#308) 

### PR DESCRIPTION
**Pull request by mgordel**

### CODE REVIEW

File: yajsapi/mid-level-api/script/command.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    this.args["from"] = await this.storageProvider.upload(this.srcPath);
+  }
+}
+
+export class DownloadFile extends Transfer {
+  constructor(private storageProvider: StorageProvider, private srcPath: string, private dstPath: string) {
+    super();
+    this.args = { from: `container:${srcPath}` };
+  }
+  async before() {
```
Comment (shadeofblue) [inline yajsapi/mid-level-api/script/command.ts:92]: @mgordel why does the download happen in `before`? the file has first to be transferred from the runtime to GFTP and only downloaded in `after`? am I wrong?

Comment (shadeofblue) [reply yajsapi/mid-level-api/script/command.ts:92]: have you tested these transfer commands both ways? (upload and download)?

File: yajsapi/mid-level-api/script/command.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  }
+}
+
+export class Transfer extends Command {
+  constructor(protected from?: string, protected to?: string, args?: object) {
+    super("transfer", { from, to, args });
+  }
+}
+
+export class SendFile extends Transfer {
```
Comment (shadeofblue) [inline yajsapi/mid-level-api/script/command.ts:77]: @mgordel I'd change the name here to `UploadFile` _or_ change `DownloadFile` to `ReceiveFile`

iow, either we're using upload+download or send+receive ...

File: yajsapi/mid-level-api/script/command.ts (hunk 1 above)
Comment (shadeofblue) [reply yajsapi/mid-level-api/script/command.ts:92]: okay, the name `download` confused me, as discussed, those methods should probably be renamed to `publish` vs `receive`

**Review by shadeofblue — CHANGES_REQUESTED** [review]
please see my comments

File: yajsapi/mid-level-api/script/command.ts (hunk 2 above)
Comment (mgordel) [reply yajsapi/mid-level-api/script/command.ts]: ok. I changed to UploadFile

File: yajsapi/mid-level-api/script/command.ts (hunk 1 above)
Comment (mgordel) [reply yajsapi/mid-level-api/script/command.ts:97]: I change them to publish and receive

**Pull request merged**